src/v/kafka/server/tests/create_topics_test.cc

```diff
@@ -193,6 +230,8 @@ FIXTURE_TEST_EXPECTED_FAILURES(create_topics, create_topic_fixture, 2) {
     wait_for_controller_leadership().get();
 
     test_create_topic(make_req({make_topic("topic1")}));
+    test_create_non_replicated_topic(
+      model::topic("topic1"), make_req({make_topic("topic2")}));
```

Comment:
i think there needs to be tests for

* topic deletion is prevent
* topic creation for non-existent source topic
* topic creation for duplicate materialized topic name
* leadership query (matches source topic)
* metadata (e.g. partitions) for materialized topic matches source topic
* error when producing to materialized topic

it seems like this PR covers some partition movement scenario, so we could probably integrate some tests with that to verify that materialized topics are moved as well.

Reply:
Actually whatever did touch partition movement scenarios were removed. Adding all of these scenarios

Reply:
> i think there needs to be tests for
> 
> * topic deletion is prevent
> * topic creation for non-existent source topic
> * topic creation for duplicate materialized topic name
> * leadership query (matches source topic)
> * metadata (e.g. partitions) for materialized topic matches source topic
> * error when producing to materialized topic
> 
> it seems like this PR covers some partition movement scenario, so we could probably integrate some tests with that to verify that materialized topics are moved as well.

Test coverage for metadata requests working w/ materialized topics already exists in `coproc/tests/read_materialized_topic_test.cc` and its also prevelent in the `create_topics_tests*` as after each call to `create_topics` a metadata request is made to assert that the effects worked. All other listed cases are covered as of the most recent force push.